File: lib/limelight/commands/version_command.rb
#- Copyright © 2008-2010 8th Light, Inc. All Rights Reserved.
#- Limelight and all included source files are distributed under terms of the GNU LGPL.

require 'limelight/commands/command'

module Limelight
  module Commands

    # Prints the current Limelight version
    #
    class VersionCommand < Command

      install_as "--version"

      def self.description #:nodoc:
        return "Prints the current Limelight version"
      end

      protected ###########################################

      def process #:nodoc:
        require 'limelight/version'
        puts "Limelight, version #{VERSION::STRING}"
      end

    end

  end


end

File: src/limelight/commands/VersionCommand.java
package limelight.commands;

import limelight.About;

import java.util.Map;

public class VersionCommand extends Command
{
  private static Arguments arguments = new Arguments();

  @Override
  public void doExecute(Map<String, String> args)
  {
    say("limelight " + About.version);
  }

  @Override
  public String description()
  {
    return "Prints current version";
  }

  @Override
  public String name()
  {
    return "version";
  }

  @Override
  public Arguments getArguments()
  {
    return arguments;
  }
}
